codegenerator/cli/templates/static/codegen/src/eventFetching/FetchState.res

```diff
@@ -523,27 +525,12 @@ let getQueryLogger = (
   {fetchStateRegisterId, fromBlock, toBlock, contractAddressMapping}: nextQuery,
   ~logger,
 ) => {
-  let fetchStateRegister = switch fetchStateRegisterId {
-  | Root => "root"
-  | DynamicContract({blockNumber, logIndex}) =>
-    `dynamic-${blockNumber->Int.toString}-${logIndex->Int.toString}`
-  }
-  let addressesAll = contractAddressMapping->ContractAddressingMap.getAllAddresses
-  let (displayAddr, restCount) = addressesAll->Array.reduce(([], 0), (
-    (currentDisplay, restCount),
-    addr,
-  ) => {
-    if currentDisplay->Array.length < 3 {
-      (Array.concat(currentDisplay, [addr->Address.toString]), restCount)
-    } else {
-      (currentDisplay, restCount + 1)
-    }
-  })
```

Comment:
The Array.reduce could actually affect peformance for indexers with many addresses. Optimised it a little bit.